DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -83,19 +97,24 @@ void FillBindingData(
 
             for (auto& source : sources)
             {
-                assert(source.resource != nullptr);
-                assert(source.resourceDesc != nullptr);
-
-                if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                assert(source.resource != nullptr);   
+                if (isSerializedGraph)
                 {
-                    throw std::invalid_argument("DML operators only support buffer bindings");
+                    bindingData.bufferBindings[bufferIndex].SizeInBytes = source.elementCount * source.elementSizeInBytes;
+                }
+                else
+                {
+                    assert(source.resourceDesc != nullptr);
+                    if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                    {
+                        throw std::invalid_argument("DML operators only support buffer bindings");
+                    }
+
+                    auto& bufferDesc = std::get<Model::BufferDesc>(source.resourceDesc->value);
+                    bindingData.bufferBindings[bufferIndex].SizeInBytes = bufferDesc.sizeInBytes - bindingData.bufferBindings[bufferIndex].Offset;
```

Comment:
Using the Offset variable before initializing it i.e. this is being initialized at line 117 but it is being used at line 114.

Also as discussed offline, can we use the same calculation as DmlDispatchable for DmlSerializedGraph as well?

Reply:
done